transport/tchannel/handler.go

```diff
@@ -90,31 +92,39 @@ func (h handler) Handle(ctx ncontext.Context, call *tchannel.InboundCall) {
 }
 
 func (h handler) handle(ctx context.Context, call inboundCall) {
-	start := time.Now()
-	err := h.callHandler(ctx, call, start)
-	if err == nil {
-		return
+	// you MUST close the responseWriter no matter what
+	responseWriter := newResponseWriter(call.Response(), call.Format())
+
+	handlerErr := h.callHandler(ctx, call, responseWriter)
+	if handlerErr != nil {
+		// we have an error, so we're going to propagate it as a yarpc error,
+		// regardless of whether or not it is a system error.
+		yarpcError := yarpcerrors.ToYARPCError(handlerErr)
+		// TODO: what to do with error? we could have a whole complicated scheme to
+		// return a SystemError here, might want to do that
+		text, _ := yarpcerrors.ErrorCode(yarpcError).MarshalText()
+		responseWriter.addHeader(ErrorCodeHeaderKey, string(text))
+		if name := yarpcerrors.ErrorName(yarpcError); name != "" {
```

Comment:
Discussed offline. TChannel error frames should be sent for errors in our TChannel handler. Errors returned by user-implemented application handlers must not result in TChannel error frames.